esp-hal/src/soc/mod.rs

```diff
@@ -83,36 +97,40 @@ impl self::efuse::Efuse {
 }
 
 #[allow(unused)]
-pub(crate) fn is_valid_ram_address(address: u32) -> bool {
-    (self::constants::SOC_DRAM_LOW..=self::constants::SOC_DRAM_HIGH).contains(&address)
+pub(crate) fn is_valid_ram_address(address: usize) -> bool {
+    addr_in_range(address, DRAM)
 }
 
 #[allow(unused)]
 pub(crate) fn is_slice_in_dram<T>(slice: &[T]) -> bool {
-    let start = slice.as_ptr() as u32;
-    let end = start + slice.len() as u32;
-    self::constants::SOC_DRAM_LOW <= start && end <= self::constants::SOC_DRAM_HIGH
+    slice_in_range(slice, DRAM)
 }
 
 #[allow(unused)]
-pub(crate) fn is_valid_psram_address(address: u32) -> bool {
-    #[cfg(psram)]
-    {
-        let memory_range = MAPPED_PSRAM.with(|mapped_psram| mapped_psram.memory_range.clone());
-        memory_range.contains(&(address as usize))
-    }
-    #[cfg(not(psram))]
-    false
+pub(crate) fn is_valid_psram_address(address: usize) -> bool {
+    addr_in_range(address, psram_range())
 }
 
 #[allow(unused)]
 pub(crate) fn is_slice_in_psram<T>(slice: &[T]) -> bool {
-    let start = slice.as_ptr() as u32;
-    let end = start + slice.len() as u32;
```

Comment:
Ah yeah this was subtly wrong. It didn't take into account the size of `T`.

Reply:
I didn't even notice :)